### DISCUSSION TITLE
The minimum plot size will be k=32

### DISCUSSION THREAD

**Discussion by hoffmang9** [Chia-Network/chia-blockchain#452] [General]
All,

Though k=30 had been ruled out, it's now time to rule out k=31.

@wjblanke and I were able to get a k=31 phase 1 to complete in about 16 minutes on $4,000.00 equivalent hardware. That means that in two Moore's Law cycles of 18 months, a k=31 phase 1 arguably can be completed in 4 minutes. Since it's theoretically possible to farm at the end of phase 1, that's cutting it too close.

Someday the minimum k will go up from 32. However we'd like that to be more like 10 years from now rather than the 2-3 years that making k=31 minimum would have caused.

**Comment by marvic2409** [comment]
What was the hardware used in the testing?

**Reply by hasbrowow** [reply]
Bumping this 👍

**Comment by felicityc** [comment]
moore's law is dead.

it'll happen even faster.

**Reply by dorkmo** [reply]
im only doing 33s lol

**Reply by dipique** [reply]
Yes, it's dead, but we're now slower than Moore's law, not faster.

**Comment by vitiaaap29** [comment]
Hi @hoffmang9, excuse for troubling
 I can select K = 25 
<img width="944" alt="изображение" src="https://user-images.githubusercontent.com/5259072/115689977-45065780-a365-11eb-819e-23101dad456a.png">
Is it ok? Will my farming have a profit?

I use Windows 7, 4Gb RAM, 64-bit, processor: AMD A6-7310 APU with AMD Radeon R4 Graphics

**Reply by GrayXu** [reply]
plots with small k won't be accepted by mainnet

**Reply by vitiaaap29** [reply]
Thanks @GrayXu I will try to use K = 32

**Reply by SnoekieCookie** [reply]
> plots with small k won't be accepted by mainnet
What does it mean? The status indicates that the farm is in progress

**Reply by SnoekieCookie** [reply]
> plots with small k won't be accepted by mainnet

At the end of the farm, I will not receive a reward?

**Reply by ilyakooo0** [reply]
Today you will not receive rewards for k less than 32

**Reply by SnoekieCookie** [reply]
> Сегодня вы не получите награды за k меньше 32

Why is it possible to mine it? How can I use it?

**Reply by ilyakooo0** [reply]
В принципе можно генерировать плоты любого уровня сложности (k). От уровня сложности и того, как он соотносится с доступностью вычислительных мощностей, зависит уровень безопасности всей системы. Если генерация будет слишком легкой – появляются новые векторы атаки. По этой причине в валидаторах программно ограничивают то, какой уровень сложности они считают валидными. транзакции, которые генерированы со слишком маленьким k будут попросту отвергаться как небезопасные. 

На сегодняшний день 32 – это минимальный уровень сложности, который считается безопасным.

В интерфейсе он остался скорее всего просто по историческим причинам или для быстрого тестирования функциональности.

**Reply by SnoekieCookie** [reply]
> В принципе можно генерировать плоты любого уровня сложности (k). От уровня сложности и того, как он соотносится с доступностью вычислительных мощностей, зависит уровень безопасности всей системы. Если генерация будет слишком легкой – появляются новые векторы атаки. По этой причине в валидаторах программно ограничивают то, какой уровень сложности они считают валидными. транзакции, которые генерированы со слишком маленьким k будут попросту отвергаться как небезопасные.
> 
> На сегодняшний день 32 – это минимальный уровень сложности, который считается безопасным.
> 
> В интерфейсе он остался скорее всего просто по историческим причинам или для быстрого тестирования функциональности.

А насколько вероятно, что в ближайшие несколько месяцев и к-32 тоже стане не безопасным и перестанет приниматься? Получается он тоже может быстро стать таким же бесполезным как и к-25?

**Reply by ilyakooo0** [reply]
по оценкам 32 будет безопасным примерно лет 10 еще.

**Reply by najchris11** [reply]
> > Сегодня вы не получите награды за k меньше 32
> 
> Why is it possible to mine it? How can I use it?

I actually have this same question. Should i get a bigger temp drive and focus on k=32?

**Reply by ilyakooo0** [reply]
@najchris11 For sure. All your transaction with k less than 32 will be rejected and you will not get any rewards.

**Reply by najchris11** [reply]
ok cool! thanks!

**Reply by stdiv** [reply]
> plots with small k won't be accepted by mainnet

Привет! Подскажите плиз, есть ли сейчас смысл делать плоты более чем 32, за них награда будет начисляться? вроде говорили что у 32 и 33 разная вероятность выигрыша - так ли то? Спасибо

**Reply by davecarter** [reply]
The bigger the plot size the more time will it take to farm. As an advantage for the effort, it will not become obsolete very easily so it will be online longer.

**Reply by hoffmang9** [reply]
Larger plots do not take more time to farm. The only variable that matters is total space plotted.

**Reply by stdiv** [reply]
Tell me how the common space is determined, if I have 2 computers in different networks, full nodes are running (the chia application is running), everything works under one wallet, but they do not see each other's rafts, on one 100 on the other 30. Is it possible to do this and depends whether from this the likelihood of closing the block. Thank you
На русском: Подскажите как определяется общее пространство, если у меня 2 компьютера в разных сетях, запущены полные ноды ( приложение chia запущено) все работает под одним кошельком, но плоты друг друга они не видят, на одном 100 на другом 30. Можно ли так делать и зависит ли от этого вероятность закрыть блок. Спасибо

**Comment by Icecapital** [comment]
Помогите разобраться, для того что бы начать фармить на диске должно быть свободно 239 гигабайт для построения и в конечном итоге занято будет 101.4 ?  Или достаточно просто иметь 101.4 гб для K =32, у меня есть свободные диски 240гб

**Reply by ilyakooo0** [reply]
239 для построения и 101.4 в итоге

**Reply by giobit** [reply]
> 239 для построения и 101.4 в итоге

для посева участка размером 101.4 гиб  тибе нужно иметь  на быстром носителе SSD типа М2 свободное пространство 239 гиб это для создания на нём кэша. например если утебя М2  512гб  сможешь посеить только один участок  и после посева опять один и так далее если утебя М2   размером 1ТБ   сможешь посеить 3 участка паралельно. временной кэш рассчитывай  так  (K=32- 269 GB (288.8 GB) ) это был пункт 3.  пункт 4 где уже из кэша переносица на основной носитель жостки диск (HDD) чем больше тем лудше чтобы много посевов было. после этого твой М2 не нужен он не участвуит в фарминге.  главное иметь быстрый ссд для посева ото будешь сеит  весь год :Д а там сложность вырастит до небес а тут главное сделать всё быстро чтобы быстрее заработать.

**Comment by ChandlerFerry** [comment]
@hoffmang9 With k32 plots being completed in sub 10 minutes due to a new multithreaded plotter (https://github.com/madMAx43v3r/chia-plotter). Will Chia-Network be moving to k34 plots to maintain the 10 year buffer?

**Reply by hoffmang9** [reply]
No

**Reply by ChandlerFerry** [reply]
Interesting, and exceptionally good to know for the longevity of Chia.

Out of curiosity, what allows you to say this with certainty when there's a risk of an attack vector being approached?

**Reply by ChandlerFerry** [reply]
Follow up to that, on https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#what-k-size-should-you-plot there's a statement: `The goal is to keep it so that the top-of-the-line hardware takes at least 1 hour to plot the minimum k-size, and Phase 1 takes at least ~10 minutes so there is no way to cheat the system.`

As both of these have already been overtaken, should this get edited to be more transparent?

**Reply by hoffmang9** [reply]
It's an older comment about the original and not new consensus.

**Reply by 69chargr** [reply]
I've tested this and see little improvement over using the combined chiapos mod along with Swar's. I've been able to crank out 43 ppd so far.

I managed a 1969 second K32 plot on my 5900x with this new plotter. Seems amazing until I divide 86,400 by 1969 and get.... 43 :-)

```
Plot Name: plot-k32-2021-06-09-00-09-d24436b4bd4f0871b26a68eafb3fa60769a1d3d35f5e506028b34ae34047cdd0
[P1] Table 1 took 12.2591 sec
[P1] Table 2 took 96.7521 sec, found 4294950613 matches
[P1] Table 3 took 154.236 sec, found 4294871360 matches
[P1] Table 4 took 193.88 sec, found 4294970466 matches
[P1] Lost 2857 matches due to 32-bit overflow.
[P1] Table 5 took 190.932 sec, found 4294968294 matches
[P1] Lost 745 matches due to 32-bit overflow.
[P1] Table 6 took 177.279 sec, found 4294993872 matches
[P1] Lost 26367 matches due to 32-bit overflow.
[P1] Table 7 took 122.585 sec, found 4295042485 matches
Phase 1 took 947.996 sec
[P2] max_table_size = 4295042485
[P2] Table 7 scan took 13.0649 sec
[P2] Table 7 rewrite took 37.295 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 28.7185 sec
[P2] Table 6 rewrite took 52.859 sec, dropped 581269867 entries (13.5337 %)
[P2] Table 5 scan took 26.8092 sec
[P2] Table 5 rewrite took 50.0234 sec, dropped 761953248 entries (17.7406 %)
[P2] Table 4 scan took 25.6456 sec
[P2] Table 4 rewrite took 49.4492 sec, dropped 828870553 entries (19.2986 %)
[P2] Table 3 scan took 25.3471 sec
[P2] Table 3 rewrite took 49.0754 sec, dropped 855036472 entries (19.9083 %)
[P2] Table 2 scan took 26.2195 sec
[P2] Table 2 rewrite took 48.8021 sec, dropped 865574957 entries (20.1533 %)
Phase 2 took 442.474 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 44.9643 sec, wrote 3429375656 right entries
[P3-2] Table 2 took 32.8737 sec, wrote 3429375656 left entries, 3429375656 final
[P3-1] Table 3 took 50.5931 sec, wrote 3439834888 right entries
[P3-2] Table 3 took 33.6685 sec, wrote 3439834888 left entries, 3439834888 final
[P3-1] Table 4 took 51.9865 sec, wrote 3466099913 right entries
[P3-2] Table 4 took 33.1702 sec, wrote 3466099913 left entries, 3466099913 final
[P3-1] Table 5 took 52.1563 sec, wrote 3533015046 right entries
[P3-2] Table 5 took 34.0539 sec, wrote 3533015046 left entries, 3533015046 final
[P3-1] Table 6 took 55.4397 sec, wrote 3713724005 right entries
[P3-2] Table 6 took 34.3975 sec, wrote 3713724005 left entries, 3713724005 final
[P3-1] Table 7 took 62.5906 sec, wrote 4295042485 right entries
[P3-2] Lost 75189 entries due to 32-bit overflow.
[P3-2] Table 7 took 53.3106 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 541.674 sec, wrote 21877016804 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 36.839 sec, final plot size is 108834963207 bytes
Total plot creation time was 1969.03 sec
```

I tried running two jobs in parallel and it just took twice as long per plot. So in very limited cases, where you needed to run a single job in ~30min, I don't see where this is a big deal so far.

**Reply by Forusim** [reply]
> It's an older comment about the original and not new consensus.

Please elaborate, what is the current threshold for phase 1. We need transparency here, since the replotting for pools is imminent. Chia Inc. failed to predict the optimization potential for chiapos already. People will rage out, if Chia Inc. make k33/34 the new minimum in 5 weeks instead of 5 years.

**Reply by madMAx43v3r** [reply]
The way I understand it, unless it's cheaper to create a k32 on the fly compared to having it sit on a HDD for the time it takes, there is no point. And we are very very far away from that.

(I'm the author of said plotter)

**Reply by madMAx43v3r** [reply]
As a comparison, I can do about 40 k32 plots a day on my R720, drawing ~400W of power. That's about 4 TiB of plots for 400W, compared to a 8 TiB HDD drawing only ~5W, so it's more than 100 times more expensive. Not counting hardware cost itself, which is a lot more too.

**Reply by BitSec01** [reply]
@madMAx43v3r And lower energy usage is exact what Chia stands for right xD I guess you plotter does 2 in 1 faster and more energy efficient.

**Reply by madMAx43v3r** [reply]
Also burns less SSDs, if you use RAM for `tmpdir2`, about 75% less total writes.

**Reply by 69chargr** [reply]
Did more testing. Yeah I bottlenecked my NVMe raid 0 so hard trying to overlap phases 1 & 3 on parallel processes that my CPU usage started to plummet. So for me at least it has zero advantage over pechy combined + Swar.

**Reply by SkaZun** [reply]
@madMAx43v3r did a great job, I assigned both temp directories to the ramdisk and got very nice results, but now temp files are the bottleneck... 
need an in-memory plotter implementation )
```
Number of Threads: 42
Number of Sort Buckets: 2^7 (128)
Working Directory:   /mnt/ramdisk/
Working Directory 2: /mnt/ramdisk/
Plot Name: plot-k32-2021-06-09-16-38-3a574253bb701f63781e1f0786022afef3cd3337345247dad149d55c0f18bf1e
[P1] Table 1 took 9.98027 sec
[P1] Table 2 took 67.9665 sec, found 4294954889 matches
[P1] Table 3 took 77.4056 sec, found 4294987507 matches
[P1] Lost 20119 matches due to 32-bit overflow.
[P1] Table 4 took 94.4779 sec, found 4295127774 matches
[P1] Lost 160312 matches due to 32-bit overflow.
[P1] Table 5 took 92.8722 sec, found 4295116156 matches
[P1] Lost 148445 matches due to 32-bit overflow.
[P1] Table 6 took 91.5166 sec, found 4295057741 matches
[P1] Lost 90254 matches due to 32-bit overflow.
[P1] Table 7 took 72.0964 sec, found 4295059486 matches
Phase 1 took 506.321 sec
[P2] max_table_size = 4295127774
[P2] Table 7 scan took 6.69094 sec
[P2] Table 7 rewrite took 31.2909 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 11.7338 sec
[P2] Table 6 rewrite took 28.5968 sec, dropped 581331506 entries (13.5349 %)
[P2] Table 5 scan took 11.5081 sec
[P2] Table 5 rewrite took 27.5936 sec, dropped 762085681 entries (17.7431 %)
[P2] Table 4 scan took 11.4127 sec
[P2] Table 4 rewrite took 27.2337 sec, dropped 828984582 entries (19.3006 %)
[P2] Table 3 scan took 11.3709 sec
[P2] Table 3 rewrite took 27.2352 sec, dropped 855066740 entries (19.9085 %)
[P2] Table 2 scan took 11.2932 sec
[P2] Table 2 rewrite took 27.4581 sec, dropped 865534198 entries (20.1523 %)
Phase 2 took 248.377 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 45.5423 sec, wrote 3429420691 right entries
[P3-2] Table 2 took 51.024 sec, wrote 3429420691 left entries, 3429420691 final
[P3-1] Table 3 took 29.1082 sec, wrote 3439920767 right entries
[P3-2] Table 3 took 49.9151 sec, wrote 3439920767 left entries, 3439920767 final
[P3-1] Table 4 took 29.5414 sec, wrote 3466143192 right entries
[P3-2] Table 4 took 51.1044 sec, wrote 3466143192 left entries, 3466143192 final
[P3-1] Table 5 took 29.4716 sec, wrote 3533030475 right entries
[P3-2] Table 5 took 49.5622 sec, wrote 3533030475 left entries, 3533030475 final
[P3-1] Table 6 took 30.7462 sec, wrote 3713726235 right entries
[P3-2] Table 6 took 54.191 sec, wrote 3713726235 left entries, 3713726235 final
[P3-1] Table 7 took 53.05 sec, wrote 4295059486 right entries
[P3-2] Lost 92190 entries due to 32-bit overflow.
[P3-2] Table 7 took 60.8588 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 538.532 sec, wrote 21877208656 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 61.6404 sec, final plot size is 108835749246 bytes
Total plot creation time was 1354.93 sec
```

**Reply by madMAx43v3r** [reply]
Not bad. There's a couple bottlenecks in phase 3 for such a beasts of CPU, I'm working on it. Complete in-memory, I was thinking about it, it's quite a bit of work..

**Reply by Ryzripper** [reply]
> Not bad. There's a couple bottlenecks in phase 3 for such a beasts of CPU, I'm working on it. Complete in-memory, I was thinking about it, it's quite a bit of work..

Will this plotter be for k-33 plots?

**Reply by dorkmo** [reply]
about 3% of netspace is k33. so some people are planning for the increase, but it will be interesting to see if chia hq plan on incrementally going up by one, or will they stick to even numbers or something.

**Reply by winkee01** [reply]
> Not bad. There's a couple bottlenecks in phase 3 for such a beasts of CPU, I'm working on it. Complete in-memory, I was thinking about it, it's quite a bit of work..

keep going, you are a real genius

**Reply by chefsas** [reply]
> 
> 
> Not bad. There's a couple bottlenecks in phase 3 for such a beasts of CPU, I'm working on it. Complete in-memory, I was thinking about it, it's quite a bit of work..

Thank you for everything you're doing for the community.

-Scott

**Reply by bramcohen** [reply]
On the question of risk to k=32 it isn't enough to simply get the proofs in fast for k32 to be in danger it must also be that it's more cost effective to have that hardware generating plots on the fly against the existing netspace. A few things complicate this: You only need the first phase to be done and you get a factor of 512 because of the filters, but you're competing against well below cost disks at equilibrium. Even with those factors it still appear k32 is quite safe, and if it become unsafe for plotting reasons it isn't clear that cranking up the minimum k value will be the right countermeasure.

**Reply by ChandlerFerry** [reply]
While I understand the argument about cost vs reward, I do not believe that is the right approach for thinking about how to secure a blockchain. A blockchain's security should depend on the hardware that is currently available, but not on whether or not it's profitable to exploit it. All cryptocurrency prices change daily, and a bad actor could gamble on Chia's price doubling over the next two years.


I completely agree that if it becomes unsafe for plotting reasons cranking up the minimum k value may not be the right countermeasure. **However, I would love to know what other options there are for the future of Chia.**

I say this, because my understanding is that the bottleneck is currently RAM speed (or OpenGL/OpenCL Plotting):
- k32 is unsafe because of DDR5 RAM
- k33 may be unsafe due to DDR6 RAM

**Reply by madMAx43v3r** [reply]
Well well... a factor of 512, how did I miss that... That changes the game quite a bit, now you got my GPU brain thinking. Some rough calculations tell me it's a close race, I think only an attempted implementation could tell if it's more profitable or not. (If a decent CPU can do phase 1 in 10 min, a decent GPU should do it in ~15 sec, assuming BLAKE3 plays well on GPUs. Such a GPU will probably draw ~250W and it competes against 1024 plots on lets say ten 10TB HDDs, drawing ~100W.)

**Reply by delucca** [reply]
IMHO we should find a way to use @madMAx43v3r plotter as the Chia default plotter :)

It is pretty impressive the work he did!

**Reply by delucca** [reply]
> It's an older comment about the original and not new consensus.

Great! I was afraid of that too. Do plotting time is still a security breach for the new consensus algorithm? Or even creating plots in 1s would not be a feasible attack?

**Reply by hoffmang9** [reply]
We are absolutely game to merge them together.

**Reply by madMAx43v3r** [reply]
I guess we can add it as a special case for k32, via a submodule without the bls dependency.

**Reply by bramcohen** [reply]
We can't merge in the madmax plotter under the current license of GPLv3. If it's released Apache 2 we'd be very interested in merging it as the default, with the caveat that we need to try a bunch of different platforms/hardware/circumstances and see what works best on each.

**Reply by madMAx43v3r** [reply]
The license can be changed, no issues there.

**Reply by ewan1983** [reply]
> Well well... a factor of 512, how did I miss that... That changes the game quite a bit, now you got my GPU brain thinking. Some rough calculations tell me it's a close race, I think only an attempted implementation could tell if it's more profitable or not. (If a decent CPU can do phase 1 in 10 min, a decent GPU should do it in ~15 sec, assuming BLAKE3 plays well on GPUs. Such a GPU will probably draw ~250W and it competes against 1024 plots on lets say ten 10TB HDDs, drawing ~100W.)

Does that mean GPU can solve any puzzle in 15 seconds, and winning every reward?

**Reply by xorinox** [reply]
> 
> 
> I've tested this and see little improvement over using the combined chiapos mod along with Swar's. I've been able to crank out 43 ppd so far.
> 
> I managed a 1969 second K32 plot on my 5900x with this new plotter. Seems amazing until I divide 86,400 by 1969 and get.... 43 :-)
> 
> ```
> Plot Name: plot-k32-2021-06-09-00-09-d24436b4bd4f0871b26a68eafb3fa60769a1d3d35f5e506028b34ae34047cdd0
> [P1] Table 1 took 12.2591 sec
> [P1] Table 2 took 96.7521 sec, found 4294950613 matches
> [P1] Table 3 took 154.236 sec, found 4294871360 matches
> [P1] Table 4 took 193.88 sec, found 4294970466 matches
> [P1] Lost 2857 matches due to 32-bit overflow.
> [P1] Table 5 took 190.932 sec, found 4294968294 matches
> [P1] Lost 745 matches due to 32-bit overflow.
> [P1] Table 6 took 177.279 sec, found 4294993872 matches
> [P1] Lost 26367 matches due to 32-bit overflow.
> [P1] Table 7 took 122.585 sec, found 4295042485 matches
> Phase 1 took 947.996 sec
> [P2] max_table_size = 4295042485
> [P2] Table 7 scan took 13.0649 sec
> [P2] Table 7 rewrite took 37.295 sec, dropped 0 entries (0 %)
> [P2] Table 6 scan took 28.7185 sec
> [P2] Table 6 rewrite took 52.859 sec, dropped 581269867 entries (13.5337 %)
> [P2] Table 5 scan took 26.8092 sec
> [P2] Table 5 rewrite took 50.0234 sec, dropped 761953248 entries (17.7406 %)
> [P2] Table 4 scan took 25.6456 sec
> [P2] Table 4 rewrite took 49.4492 sec, dropped 828870553 entries (19.2986 %)
> [P2] Table 3 scan took 25.3471 sec
> [P2] Table 3 rewrite took 49.0754 sec, dropped 855036472 entries (19.9083 %)
> [P2] Table 2 scan took 26.2195 sec
> [P2] Table 2 rewrite took 48.8021 sec, dropped 865574957 entries (20.1533 %)
> Phase 2 took 442.474 sec
> Wrote plot header with 268 bytes
> [P3-1] Table 2 took 44.9643 sec, wrote 3429375656 right entries
> [P3-2] Table 2 took 32.8737 sec, wrote 3429375656 left entries, 3429375656 final
> [P3-1] Table 3 took 50.5931 sec, wrote 3439834888 right entries
> [P3-2] Table 3 took 33.6685 sec, wrote 3439834888 left entries, 3439834888 final
> [P3-1] Table 4 took 51.9865 sec, wrote 3466099913 right entries
> [P3-2] Table 4 took 33.1702 sec, wrote 3466099913 left entries, 3466099913 final
> [P3-1] Table 5 took 52.1563 sec, wrote 3533015046 right entries
> [P3-2] Table 5 took 34.0539 sec, wrote 3533015046 left entries, 3533015046 final
> [P3-1] Table 6 took 55.4397 sec, wrote 3713724005 right entries
> [P3-2] Table 6 took 34.3975 sec, wrote 3713724005 left entries, 3713724005 final
> [P3-1] Table 7 took 62.5906 sec, wrote 4295042485 right entries
> [P3-2] Lost 75189 entries due to 32-bit overflow.
> [P3-2] Table 7 took 53.3106 sec, wrote 4294967296 left entries, 4294967296 final
> Phase 3 took 541.674 sec, wrote 21877016804 entries to final plot
> [P4] Starting to write C1 and C3 tables
> [P4] Finished writing C1 and C3 tables
> [P4] Writing C2 table
> [P4] Finished writing C2 table
> Phase 4 took 36.839 sec, final plot size is 108834963207 bytes
> Total plot creation time was 1969.03 sec
> ```
> 
> I tried running two jobs in parallel and it just took twice as long per plot. So in very limited cases, where you needed to run a single job in ~30min, I don't see where this is a big deal so far.

The amazing isn't the speed or the hope that is works even faster than the reference plotter (chiapos), the amazing is, that this can be done with fewer SSDs.  It will allow for a much cheaper plotter. The issues with ramdisk were so far,  you needed really a lot so you can go all parallel process plotting. Now you only need one smallish one.  A plotter with one decent SSD (temp1) and 110G of ramdisk for (temp2) is now and could soon become the desired plotter design..

**Reply by slothtechtv** [reply]
I agree it should be merged if possible. As far as testing on different hardware configurations, I've done a fair bit myself. So far Madmax has worked well with everything I've thrown at it: consumer NVMe, ram, enterprise NVMe, enterprise SAS SSD, and SAS magnetic drives (though I agree more testing is needed).

I've gotten plot times below 1400 seconds with enough compute behind it. I think the real benefit to this plotter is not the speed; although it is impressive, the real benefit is how it levels the playing field and encourages the use of older hardware.

I think this plotter makes it more equitable for people to plot and fill up whatever space they have, because people can maximize their machines without expensive NVMe drives, and as an added bonus there is less of a concern of endurance or "waste." I've gotten great performance while running this plotter with lots of threads against old 300G SAS 10K drives as both TMP1 and TMP2 -- no ram or SSD needed at all (besides what's used to run the plotter...)

The value proposition alone brings so much utility. While the stock plotter gets the job done, it really limits your options for generating plots. You really only have one option to fully utilize a high thread count CPU-- you have to leverage lots and lots of flash storage since most of your compute will be idle without enough of it. By being able to fully saturate all of the threads on even a high thread count CPU with only a few drives (even old 10K SAS drives) so many doors open up, and with them comes flexibility.

**Reply by exander77** [reply]
> Well well... a factor of 512, how did I miss that... That changes the game quite a bit, now you got my GPU brain thinking. Some rough calculations tell me it's a close race, I think only an attempted implementation could tell if it's more profitable or not. (If a decent CPU can do phase 1 in 10 min, a decent GPU should do it in ~15 sec, assuming BLAKE3 plays well on GPUs. Such a GPU will probably draw ~250W and it competes against 1024 plots on lets say ten 10TB HDDs, drawing ~100W.)

Do I understand correctly that you can make a plot that passes a filter? Hence scale 512x? And if you do it twice per block period (had to be more like 10s each), scale 1024x, effectively putting GPU to compare with roughly 120GB of plots? That's almost 1PB for 8x GPU rig.

This sounds to me like Chia end-game. We are talking about today's GPUs.

**Reply by xorinox** [reply]
Still pretty far out in the future and if that day comes,  the storage market doesn't sleep.  You need to consider power draw.  The 18 TB WD DC550 uses about 6.5 watts operating (WD spec).  1,048,576 GiB / 101.366 GiB/K32 = 10,345 plots rounded up,  divide by 165 plots per 18 TB HDD = 63x 18 TB HDD rounded up.  63 times 6.5 watts per 18TB HDD = 410 watts rounded up.  Let's add 90 watts for a system with Ryzen 7 3700x, motherboard and HBAs to run the drives. Let's add another 100 watts to compensate likely overly optimistic WD spec as for watts per HDD..  600 watts for 1 PiB of plots.  I skipped the part of how to operate 63 HDD, maybe 2x 44 bay JBODs that if they are empty use 100 watts each (fans and back plane?)  800 watts, how many GPUs can you run with that?  800 watts/PiB

The setup discussed will be a server and 2x JBODs.  Not too difficult of to a setup.  Or you build a farming rig from scratch, like the RockPI4 farmer:  https://github.com/Chia-Network/chia-blockchain/wiki/Reference-Farming-Hardware the rig as of now home for 3,983 plots uses 225 watts or 5.22 kWh/day on average for the last 30 days (real time measured), that is 585 watts/PiB  if I am not mistaken.

Tomorrow or sometime in the future when real time compute plot might be possible,  GPUs will be more efficient, CPUs also, but hard drives will be bigger too (less power per TB).  And a little after that we will see HDD getting real competition from low cost SSD for data storage. When we get there, power consumption to store data will drop dramatically.

**Reply by exander77** [reply]
@xorinox It may have been an old consensus, but 1H and 10M were milestones for speed and We may be already passed them.

The problem I see is that you can't really improve proof of space efficiency in any major way. You can squeeze 10-15% from compression. You can improve the power efficiency and capacity of drives, but you are seriously limited based on the progress.

On the other hand, for GPU estimate is just preliminary. And I have witnessed GPU improvements over the years closely. Speeding out calculation twice either with raw power or algorithm improvements is usually quite fast. Doubling capacities will be much much harder. I don't see HDD development keeping up with GPU development. If somebody brings it down to 5s a plot on GPU, it will be effectively equivalent to 2560x k=32 plots. This could be at the end of 2022.

**Reply by madMAx43v3r** [reply]
An effective countermeasure would be to make more plots pass the filter, instead of 1/512 like 1/128. Right now the amount of disk access is really low, a 4x or even 8x increase shouldn't be an issue.

**Reply by xorinox** [reply]
> 
> 
> An effective countermeasure would be to make more plots pass the filter, instead of 1/512 like 1/128. Right now the amount of disk access is really low, a 4x or even 8x increase shouldn't be an issue.

like not really relevant but there used to be Chia days without filter :-)

**Reply by ddowhy** [reply]
> > Well well... a factor of 512, how did I miss that... That changes the game quite a bit, now you got my GPU brain thinking. Some rough calculations tell me it's a close race, I think only an attempted implementation could tell if it's more profitable or not. (If a decent CPU can do phase 1 in 10 min, a decent GPU should do it in ~15 sec, assuming BLAKE3 plays well on GPUs. Such a GPU will probably draw ~250W and it competes against 1024 plots on lets say ten 10TB HDDs, drawing ~100W.)
> 
> Do I understand correctly that you can make a plot that passes a filter? Hence scale 512x? And if you do it twice per block period (had to be more like 10s each), scale 1024x, effectively putting GPU to compare with roughly 120GB of plots? That's almost 1PB for 8x GPU rig.
> 
> This sounds to me like Chia end-game. We are talking about today's GPUs.

wouldn't the dev's just up the minimum k size to something higher than what any group of gpus would be able to theoretically do?

**Reply by bampan** [reply]
And what stops people from building asics?

**Reply by BitSec01** [reply]
> And what stops people from building asics?

memory

edit: At least that's what I've always heard is a hurdle too big for asic creators.

**Reply by exander77** [reply]
> > And what stops people from building asics?
> 
> memory
> 
> edit: At least that's what I've always heard is a hurdle too big for asic creators.

I don't think you need much memory. You have to take into account, that you will not be sorting or storing anything, you will just test if the attempt leads to proof and if not you will throw it away. The memory used for plotting is for sorting, compression and storing actual plot file.

**Reply by kanasite** [reply]
> > Well well... a factor of 512, how did I miss that... That changes the game quite a bit, now you got my GPU brain thinking. Some rough calculations tell me it's a close race, I think only an attempted implementation could tell if it's more profitable or not. (If a decent CPU can do phase 1 in 10 min, a decent GPU should do it in ~15 sec, assuming BLAKE3 plays well on GPUs. Such a GPU will probably draw ~250W and it competes against 1024 plots on lets say ten 10TB HDDs, drawing ~100W.)
> 
> Does that mean GPU can solve any puzzle in 15 seconds, and winning every reward?

not true, the only thing that a machine is able to compute phase 1 in less than 30secs only break the current plot filter (512 factors). In layman terms, it's just having the benefit of 512x better chance to generate plot on the fly but doesn't guarantee the best proof. However, it's a matter of concern if it's economically profitable to generate the hash on the fly, which defeat the purpose of "proof of space"

**Reply by exander77** [reply]
> > > Well well... a factor of 512, how did I miss that... That changes the game quite a bit, now you got my GPU brain thinking. Some rough calculations tell me it's a close race, I think only an attempted implementation could tell if it's more profitable or not. (If a decent CPU can do phase 1 in 10 min, a decent GPU should do it in ~15 sec, assuming BLAKE3 plays well on GPUs. Such a GPU will probably draw ~250W and it competes against 1024 plots on lets say ten 10TB HDDs, drawing ~100W.)
> > 
> > 
> > Does that mean GPU can solve any puzzle in 15 seconds, and winning every reward?
> 
> not true, the only thing that a machine is able to compute phase 1 in less than 30secs only break the current plot filter (512 factors). In layman terms, it's just having the benefit of 512x better chance to generate plot on the fly but doesn't guarantee the best proof. However, it's a matter of concern if it's economically profitable to generate the hash on the fly, which defeat the purpose of "proof of space"

Exactly as you say. If it is economically viable to produce proofs with similar probability on the fly, it basically degrades proof of space into proof of work.

**Reply by kanasite** [reply]
anyway proof of space literally meant "proof of workED" ;)
